script.js

```diff
@@ -1,12 +1,271 @@
+  var opt = document.createElement("option");
+  opt.value = episode.id;
+  opt.text = selectvalue;
+  drdEpisodes.appendChild(opt);
+}
+
+//------------------------------
+
+//---- PAGE Makers ---------------
+function make_ShowsGrid(showsData) {
+  Object.entries(allShows).forEach(function ([index, e]) {
+    makeItem_Shows(e);
+  });
+}
+
+//---------------------------------
+function searchData(keyword) {
+  if (currentShowId != 0) {
+    getEpisodesAsync(currentShowId).then((e) => {
+      let searchResult = e.filter((episode) => {
+        return (
+          episode.name.toLowerCase().includes(keyword.toLowerCase()) ||
+          episode.summary.toLowerCase().includes(keyword.toLowerCase())
+        );
+      });
+      console.log("104 : " + keyword);
+      makePageforSearchedepisodes(searchResult);
+    });
+  }
+}
+
+function bindData(dataSource) {
+  Object.entries(dataSource).forEach(function ([index, e]) {
+    makeEpisodeItem(e);
+    //makeEpisodesSelect(e);
+  });
+}
+
+function makeHeaderNav(showsData, episodesData) {
+  numResults.innerText = `Displaying ${numAllepisodes} episode(s)`;
+  mainRoot.innerHTML = "";
+  bindData(allEpisodes);
+}
+
+function makePageforSearchedepisodes(allEpisodes) {
+  numResults.innerText =
+    "Displaying " + allEpisodes.length + " / " + numAllepisodes + " episode(s)";
+  mainRoot.innerHTML = "";
+  bindData(allEpisodes);
+}
+
+function makePageforEpisodes() {
+  currentShowId.then((data) => {
+    numResults.innerText = `Displaying ${data.length} episode(s)`;
+    mainRoot.innerHTML = "";
+    bindData(data);
+  });
+}
+
+function makePageforAllepisodes(allEpisodes) {
+  numResults.innerText = `Displaying ${numAllepisodes} episode(s)`;
+  mainRoot.innerHTML = "";
+  bindData(allEpisodes);
+}
+
+function makeItem_Shows(shows) {
+  let divShowItem = document.createElement("div"); //make div for each show
+  mainRoot.appendChild(divShowItem);
+  divShowItem.className = "shows";
+
+  let h1Shows = document.createElement("h1"); // make it for episode title
+  let imgShows = document.createElement("img");
+
+  divShowItem.appendChild(h1Shows);
+  divShowItem.appendChild(imgShows);
+
+  h1Shows.innerText = shows.name;
+  imgShows.src = shows.image.medium;
+  divShowItem.insertAdjacentHTML(`beforeend`, shows.summary);
+}
+
+function makeEpisodeItem(episode) {
+  let divEpisode = document.createElement("div"); //make div for each episode
+  mainRoot.appendChild(divEpisode);
+  divEpisode.className = "episode";
+
+  let h1Episode = document.createElement("h1"); // make it for episode title
+  let imgEpisode = document.createElement("img");
+
+  divEpisode.appendChild(h1Episode);
+  divEpisode.appendChild(imgEpisode);
+
+  h1Episode.innerText =
+    episode.name + " - " + episodeFormatter(episode.season, episode.number);
+
+  var mediumSrc = "images/no-image.jpg";
+  episode.image == null ? mediumSrc : (mediumSrc = episode.image.medium);
+
```

Comment:
I like the way you wrote down your if /else 👍

Reply:
Me too :) it is very cool